### ISSUE TITLE
remove_duplicated_triangles does not consider all identical permutations

### ISSUE THREAD

**Issue by clemgaut** [isl-org/Open3D#3176]
**Describe the bug**
The documentation of the remove_duplicated_triangles function in TriangleMesh does not correspond to the behavior of the function.

**To Reproduce**
The documentation says "Function that removes duplicated triangles, i.e., removes triangles that reference the same three vertices, independent of their order"

However, the following code shows that this is not the case
```
import open 3d

test_mesh1 = open3d.geometry.TriangleMesh(vertices=open3d.utility.Vector3dVector([]), triangles=open3d.utility.Vector3iVector([[0, 1, 2], [2, 1, 0]]))
nb_unique_triangles_mesh1 = len(test_mesh1.remove_duplicated_triangles().triangles)

test_mesh2 = open3d.geometry.TriangleMesh(vertices=open3d.utility.Vector3dVector([]), triangles=open3d.utility.Vector3iVector([[0, 1, 2], [2, 0, 1]]))
nb_unique_triangles_mesh2 = len(test_mesh2.remove_duplicated_triangles().triangles)

print(nb_unique_triangles_mesh1) # prints 2: no triangle was removed even though they have the same vertices in different order
print(nb_unique_triangles_mesh2) # prints 1: 1 triangle was correctly removed
```

**Expected behavior**
I would expect the function to remove all triangles having the same vertices.
It seems that the function is actually looking at the order in which vertices are listed, and considers that triangle are different if their normals are facing in opposite directions.
So either the documentation should mention that triangles are duplicate if they have the same vertices and have the same normal direction, or the function should remove duplicated triangles based on the vertices only.

**Environment (please complete the following information):**
- Open3D version: 0.12.0

**Comment by scimad** [comment]
Agree on this one. According to [this documentation](http://www.open3d.org/docs/release/python_api/open3d.geometry.TriangleMesh.html?highlight=remove_duplicate#open3d.geometry.TriangleMesh.remove_duplicated_triangles), the removal should have been performed independent of the order. But [the implementation](https://github.com/intel-isl/Open3D/blob/313315d9790c36e22bb5bb034e9c7d7f470cdf73/cpp/open3d/geometry/TriangleMesh.cpp#L721) indeed treats clockwise and anticlockwise order of triangle vertices as different.

@yxlao , Can you suggest if we want to actually remove the triangle independent of order of vertices or fix documentation? Or we can also parameterize the function to ask if we should care for order or not if that makes any sense.

**Comment by mattj23** [comment]
I think you definitely don't want to lose the ability to distinguish between coincident triangles with different normal directions, as to many applications two triangles facing the opposite directions are not duplicates.  

My feeling is that either the documentation should be more explicit about the difference between vertex order and direction, and/or there should be an option to ignore direction as well as order.

**Comment by benjaminum** [comment]
Thanks for bringing this up. The documentation is not clear. 
We consider triangles to be oriented and therefore triangles with the same vertices but different orientation are not duplicates.

**Comment by yxlao** [comment]
Fixed in #4392
